semgrep/semgrep/target_manager_extensions.py

```diff
@@ -59,55 +50,67 @@
 # files.
 GENERIC_EXTENSIONS = [FileExtension("")]
 
-T = TypeVar("T")
-
-
-def langauge_set(s: Set[str]) -> Set[Language]:
-    # hack to make mypy accept our Set[str] as a set of Language
-    return s  # type:ignore
+PYTHON_LANGUAGES = [
+    Language("python"),
+    Language("python2"),
+    Language("python3"),
+    Language("py"),
+]
+JAVASCRIPT_LANGUAGES = [Language("javascript"), Language("jsx"), Language("js")]
+TYPESCRIPT_LANGUAGES = [Language("typescript"), Language("tsx"), Language("ts")]
+JAVA_LANGUAGES = [Language("java")]
+C_LANGUAGES = [Language("c")]
+GO_LANGUAGES = [Language("golang"), Language("go")]
+RUBY_LANGUAGES = [Language("ruby"), Language("rb")]
+PHP_LANGUAGES = [Language("php")]
+LUA_LANGUAGES = [Language("lua")]
+CSHARP_LANGUAGES = [Language("csharp"), Language("cs"), Language("C#")]
+RUST_LANGUAGES = [Language("rust"), Language("Rust"), Language("rs")]
+KOTLIN_LANGUAGES = [Language("kotlin"), Language("Kotlin"), Language("kt")]
+YAML_LANGUAGES = [Language("yaml"), Language("Yaml")]
+ML_LANGUAGES = [Language("ocaml"), Language("ml")]
+JSON_LANGUAGES = [Language("json"), Language("JSON"), Language("Json")]
+REGEX_LANGUAGES = [Language("none"), Language("regex")]
+GENERIC_LANGUAGES = [Language("generic")]
 
 
 # don't use this directly because it won't be O(1) lookup
 # it's just for human readability, we process it below
-_LANGS_TO_EXTS_INTERNAL: List[Tuple[Set[Language], List[FileExtension]]] = [
-    (langauge_set({"python", "python2", "python3", "py"}), PYTHON_EXTENSIONS),
-    (langauge_set({"js", "jsx", "javascript"}), JAVASCRIPT_EXTENSIONS),
-    (langauge_set({"ts", "tsx", "typescript"}), TYPESCRIPT_EXTENSIONS),
-    (langauge_set({"java"}), JAVA_EXTENSIONS),
-    (langauge_set({"c"}), C_EXTENSIONS),
-    (langauge_set({"go", "golang"}), GO_EXTENSIONS),
-    (langauge_set({"ml", "ocaml"}), ML_EXTENSIONS),
-    (langauge_set({"rb", "ruby"}), RUBY_EXTENSIONS),
-    (langauge_set({"php"}), PHP_EXTENSIONS),
-    (langauge_set({"json", "JSON", "Json"}), JSON_EXTENSIONS),
-    (langauge_set({"lua"}), LUA_EXTENSIONS),
-    (langauge_set({"cs", "csharp", "C#"}), CSHARP_EXTENSIONS),
-    (langauge_set({"rs", "rust", "Rust"}), RUST_EXTENSIONS),
-    (langauge_set({"kt", "kotlin", "Kotlin"}), KOTLIN_EXTENSIONS),
-    (langauge_set({"yaml", "Yaml"}), YAML_EXTENSIONS),
-    (REGEX_ONLY_LANGUAGE_KEYS.union({GENERIC_LANGUAGE}), GENERIC_EXTENSIONS),
+_LANGS_TO_EXTS_INTERNAL: List[Tuple[List[Language], List[FileExtension]]] = [
```

Comment:
This should make generating the tags and extensions information here much easier: https://semgrep.dev/docs/status/